src/types/mod.rs

```diff
@@ -144,7 +145,7 @@ create_relation!(f32, PhysicalType::Primitive(PrimitiveType::Float32));
 create_relation!(f64, PhysicalType::Primitive(PrimitiveType::Float64));
 
 /// The in-memory representation of the DayMillisecond variant of arrow's "Interval" logical type.
-#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, Hash)]
+#[derive(Debug, Copy, Clone, Default, Eq, Hash)]
```

Comment:
The derived PartialEq is actually not correct because it's not taking ms overflow into account.

Reply:
If we remove the `PartialEq` from here, I think we should also remove the `Eq`, since they should agree (and this makes them to disagree).